tests/unit_tests/tools/test_psyclone.py

```diff
@@ -3,311 +3,402 @@
+                     psy_file=Path("psy_file"),
+                     alg_file=Path("alg_file"),
+                     transformation_script=lambda x, y: Path('script_called'),
+                     kernel_roots=["root1", "root2"],
+                     additional_parameters=["-c", "psyclone.cfg"])
+
+    assert call_list(fake_process) == [
+        version_command, process_command
+    ]
+
+
+@mark.parametrize("version", ["2.4.0", "2.5.0"])
+def test_process_no_api_old_psyclone(version: str,
+                                     fake_process: FakeProcess) -> None:
+    """
+    Tests unspecified API for PSyclone 2.5.0 and older.
+    """
+    version_command = ['psyclone', '--version']
+    fake_process.register(version_command,
+                          stdout='PSyclone version: ' + version)
+    process_command = ['psyclone', '-api', 'nemo', '-opsy', 'psy_file',
+                       '-l', 'all', '-s', 'script_called',
+                       '-c', 'psyclone.cfg', '-d', 'root1', '-d', 'root2',
+                       'x90_file']
+    fake_process.register(process_command)
+
     psyclone = Psyclone()
-    mock_result = get_mock_result(version)
-    transformation_function = mock.Mock(return_value="script_called")
-    config = mock.Mock()
-
-    with mock.patch('fab.tools.tool.subprocess.run',
-                    return_value=mock_result) as tool_run:
-        psyclone.process(config=config,
-                         api="",
-                         x90_file="x90_file",
-                         transformed_file="psy_file",
-                         transformation_script=transformation_function,
-                         kernel_roots=["root1", "root2"],
-                         additional_parameters=["-c", "psyclone.cfg"])
-    tool_run.assert_called_with(
-        ['psyclone', '-api', 'nemo', '-opsy', 'psy_file', '-l', 'all',
-         '-s', 'script_called', '-c',
-         'psyclone.cfg', '-d', 'root1', '-d', 'root2', 'x90_file'],
-        capture_output=True, env=None, cwd=None, check=False)
-
-
-@pytest.mark.parametrize("version", ["2.4.0", "2.5.0"])
-def test_psyclone_process_nemo_api_old_psyclone(version):
-    '''Test running old-style PSyclone (2.5.0 and earlier) when requesting
-    to transform existing files by specifying the nemo api.
-    '''
+    config = Mock()
+
+    psyclone.process(config=config,
+                     api="",
+                     x90_file=Path("x90_file"),
+                     transformed_file=Path("psy_file"),
+                     transformation_script=lambda x, y: Path('script_called'),
+                     kernel_roots=["root1", "root2"],
+                     additional_parameters=["-c", "psyclone.cfg"])
+
+    assert call_list(fake_process) == [
+        version_command, process_command
+    ]
+
+
+@mark.parametrize("version", ["2.4.0", "2.5.0"])
+def test_process_nemo_api_old_psyclone(version: str,
+                                       fake_process: FakeProcess) -> None:
```

Comment:
Why are these 2 versions tested in a separate function?  The code is just the same as that above.